### DISCUSSION TITLE
Autocomplete only for top-level module? (e.g submodul paramiko.client not suggested after importing paramiko)

### DISCUSSION THREAD

**Discussion by qupfer** [microsoft/pylance-release#5288] [Q&A]
Sorry, if I missuse the name module/class and "top-level" or something like that, I'm not that familar with this.

However,
I have some struggle with the autocomplete function in vscode.
For example, if I `import paramiko` the autocomplete function suggest only all classes from the top-level modul `paramiko`,
but not from the submodule `paramiko.client` (not even the submodule itself)

How can I enable a "deeper" autocompletion?

**Comment by brettcannon** [comment]
> or example, if I `import paramiko` the autocomplete function suggest only all classes from the top-level modul `paramiko`,
> but not from the submodule `paramiko.client` (not even the submodule itself)

Did you import the submodule?

**Reply by bschnurr** [reply]
Setting all of the index depths can be done like so:

```json
    "python.analysis.packageIndexDepths": [
        {
            "name": "",
            "depth": 20
        }
    ]
```

There's also a limit to how many files we will index. So, changing the depth probably means that value needs to be set too:

```json
    "python.analysis.userFileIndexingLimit": 2000
```


you can also set specific packages

```json
"python.analysis.packageIndexDepths": [
        {
            "name": "sktime",
            "depth": 99,
            "includeAllSymbols": true
        },
        {
            "name": "rich",
            "depth": 99
        },
        {
            "name": "statsmodels",
            "depth": 99
        },
        {
            "name": "sklearn",
            "depth": 99
        },
        {
            "name": "matplotlib",
            "depth": 99
        },
        {
            "name": "scipy",
            "depth": 2
        }
    ],
```

**Reply by qupfer** [reply]
No, I did not.
If I do that, it works for that submodule, thats an improvement. Thanks.
However, in the interactive shell or pycharm (community edition) it works without importing the submodule. 

I did also the additions bschnurr mentioned, but could not see any difference.

**Comment by bschnurr** [comment]
can you provide a code example. i think i'm missing understanding..

you dont mean just completion suggestions in the import statement, right?
![image](https://github.com/microsoft/pylance-release/assets/1946977/450af91a-e8f6-4147-9c86-7f4b6522f804)

**Reply by qupfer** [reply]
Yes, I don't mean sugestions for import. Thats that work and I can use as a "workaround". Check via import what submodules exit, import them and after that, it will work and suggest things of that submodule during code writing too.

But I would prefer an behaviour like pycharm or the tab-completion from the interactive python shell. 

![Bildschirmfoto vom 2023-12-20 13-09-27](https://github.com/microsoft/pylance-release/assets/2614431/348a3dcc-28d1-4d83-9d4a-198e6b7bbb35)
![Bildschirmfoto vom 2023-12-20 13-10-43](https://github.com/microsoft/pylance-release/assets/2614431/749d81c4-f950-4e5e-b6ae-9ea79f091857)


Edit: and thanks a lot for your help and support


Edit 2:
or is it  related to some "shortcuts" implremented?
``` 
>>> import paramiko
>>> a=paramiko.SSHClient()
>>> b=paramiko.client.SSHClient()
>>> type(a)
<class 'paramiko.client.SSHClient'>
>>> type(b)
<class 'paramiko.client.SSHClient'>
``` 
And as "a" is shown, maybe b is skipped?
